crates/swc_ecma_minifier/src/compress/pure/strings.rs

```diff
@@ -258,13 +264,15 @@ impl Pure<'_> {
                 match *e {
                     Expr::Lit(Lit::Str(s)) => {
                         cur_raw.push_str(&convert_str_value_to_tpl_raw(&s.value));
-                        cur_cooked.push_str(&convert_str_value_to_tpl_cooked(&s.value));
+                        if let Some(cur_cooked) = &mut cur_cooked {
+                            cur_cooked.push_str(&convert_str_value_to_tpl_cooked(&s.value));
+                        }
                     }
                     _ => {
                         quasis.push(TplElement {
                             span: DUMMY_SP,
                             tail: true,
-                            cooked: Some(Atom::from(&*cur_cooked)),
+                            cooked: cur_cooked.take().map(From::from),
                             raw: take(&mut cur_raw).into(),
                         });
```

Comment:
We need to reinitialize `cooked = Some(String::new())` here.

Reply:
Sorry! I missed the point. And thanks!